scripts/build-package.js

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+'use strict';
+
+require('hard-rejection/register');
+const fs = require('fs');
+const path = require('path');
+const del = require('del');
+const execa = require('execa');
+const cpy = require('cpy');
+const makeDir = require('make-dir');
+
+const rootDir = path.resolve(__dirname, '..');
+const outputDir = path.resolve(rootDir, 'pkg');
+const srcDir = path.resolve(rootDir, 'src/components');
+
+// Build a single directory that is ready to be published as an npm package.
+// From that package components should be importable without any
+// additional nesting.
+//
+// The root package.json of this repo has `"private": true` so we don't
+// accidentally publish that way. Instead, we need to publish from the
+// directory that is built by this script.
+
+// Compile all src files that we want in the package ... so that excludes
+// tests and examples.
+function compileComponents() {
+  return execa.shell(
+    `babel "**/!(examples|__tests__)/*.js" --out-dir ${outputDir}`,
+    {
+      cwd: srcDir,
+      stdio: 'inherit'
+    }
+  );
+}
+
+// Copy non-src files that we want in the package.
+function copyFiles() {
+  return cpy(['LICENSE', 'CHANGELOG.md', 'README.md'], outputDir, {
+    cwd: rootDir
+  });
+}
+
+function copyCss() {
```

Comment:
@davidtheclark this approach feels a it's maybe a bit clunky to me. Do you think it would be preferable to put `copyCss`, `copyData`, and `copyFiles` into a single function or is it ok to split them out like this?

Reply:
I tried a couple of different approaches and ran into some issues here, mostly stemming from the fact that i'd like there to be `css` and `data` directories in the finished `pkg` directory: 

- afaict, you can't specify multiple destination directories using `cpy`
- you _can_ use `cpy`'s `--parents` option to preserve directory structure, but we need to call `copyFiles()` outside of the `src` directory (see next bullet), which results in the the copied css and data files being placed into `pkg/src/css/...`, which I don't think makes sense. 
- calling `copyFiles()` from inside the `src` directory didn't work, either. I'm unsure if this is a limitation of `cpy` or something lower level, but using relative paths to parent directories with `cpy` (ex: `../README.md`) doesn't seem to work. 

As a compromise, I feel ok about combining the `copyCss()` and `copyData()` functions (see https://github.com/mapbox/dr-ui/pull/27/commits/1acb9fcee5a53b8a2040eeaa96aab23260488b03), but keeping a separate `copyFiles()` function for all of the items in the root directory. 

What do you think, @davidtheclark?